common/lib/xmodule/xmodule/vertical_block.py

```diff
@@ -0,0 +1,128 @@
+        """
+        fragment = Fragment()
+        root_xblock = context.get('root_xblock')
+        is_root = root_xblock and root_xblock.location == self.location  # pylint: disable=no-member
+
+        # For the container page we want the full drag-and-drop, but for unit pages we want
+        # a more concise version that appears alongside the "View =>" link-- unless it is
+        # the unit page and the vertical being rendered is itself the unit vertical (is_root == True).
+        if is_root or not context.get('is_unit_page'):
+            self.render_children(context, fragment, can_reorder=True, can_add=True)
+        return fragment
+
+    def get_progress(self):
+        """
+        Returns the progress on this block and all children.
+        """
+        # TODO: Cache progress or children array?
+        children = self.get_children()  # pylint: disable=no-member
+        progresses = [child.get_progress() for child in children]
+        progress = reduce(Progress.add_counts, progresses, None)
+        return progress
+
+    def get_icon_class(self):
+        """
+        Returns the highest priority icon class.
+        """
+        child_classes = set(child.get_icon_class() for child in self.get_children())  # pylint: disable=no-member
+        new_class = 'other'
+        for higher_class in CLASS_PRIORITY:
+            if higher_class in child_classes:
+                new_class = higher_class
+        return new_class
+
+    @classmethod
+    def definition_from_xml(cls, xml_object, system):
+        children = []
+        for child in xml_object:
+            try:
+                child_block = system.process_xml(etree.tostring(child, encoding='unicode'))  # pylint: disable=no-member
+                children.append(child_block.scope_ids.usage_id)
+            except Exception as exc:  # pylint: disable=broad-except
+                log.exception("Unable to load child when parsing Vertical. Continuing...")
+                if system.error_tracker is not None:
+                    system.error_tracker(u"ERROR: {0}".format(exc))
+                continue
+        return {}, children
+
+    def definition_to_xml(self, resource_fs):
+        xml_object = etree.Element('vertical')  # pylint: disable=no-member
+        for child in self.get_children():  # pylint: disable=no-member
+            self.runtime.add_block_as_child_node(child, xml_object)
+        return xml_object
+
+    @property
+    def non_editable_metadata_fields(self):
+        """
+        Gather all fields which can't be edited.
+        """
+        non_editable_fields = super(VerticalBlock, self).non_editable_metadata_fields
+        non_editable_fields.extend([
+            self.fields['due'],
+        ])
+        return non_editable_fields
+
+    def studio_view(self, context):
+        fragment = super(VerticalBlock, self).studio_view(context)
+        fragment.add_javascript('VerticalBlock = XModule.Descriptor;')
```

Comment:
Maybe a comment here so that people (like me) won't freak out when they see "Descriptor"?